### PR TITLE
Nars-ian's insatiable hunger recedes its influence in its mortal forms death

### LINKED ISSUE #53517: Nars-Ian can consume other pets while dead
image
image

### PR DESCRIPTION
**Pull request by itseasytosee**
## About The Pull Request

fixes #53517 

## Changelog
:cl: itseasytosee
fix: Nars-ian's insatiable hunger reseeds its influence in its mortal forms death
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 1]
```diff
@@ -523,17 +523,18 @@
...
-			"<span class='cult big bold'>DELICIOUS SOULS</span>")
-			playsound(src, 'sound/magic/demon_attack1.ogg', 75, TRUE)
-			narsie_act()
-			if(P.mind)
-				if(P.mind.hasSoul)
-					P.mind.hasSoul = FALSE //Nars-Ian ate your soul; you don't have one anymore
-				else
-					visible_message("<span class='cult big bold'>... Aw, someone beat me to this one.</span>")
-			P.gib()
+		if(!health <= 0)
```
Comment (bobbahbrown) [inline code/modules/mob/living/simple_animal/friendly/dog.dm:526]: move this check to before iterating through mobs in range, and fix it -- i'm asssuming you meant ``if (health > 0)``

```dm
..()
if (health <= 0)
	return
for (var/mob/living/simple_animal/pet/P in range(1, src))
	// stuff here
```

**Comment by cacogen** [comment]
Is it Nars-Ian's insatiable hunger's influence recedes after His mortal form's death?

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 2]
```diff
@@ -522,18 +522,20 @@
...
+				visible_message("<span class='warning'>[src] devours [P]!</span>", \
+				"<span class='cult big bold'>DELICIOUS SOULS</span>")
+				playsound(src, 'sound/magic/demon_attack1.ogg', 75, TRUE)
+				narsie_act()
+				if(P.mind)
+					if(P.mind.hasSoul)
+						P.mind.hasSoul = FALSE //Nars-Ian ate your soul; you don't have one anymore
+					else
+						visible_message("<span class='cult big bold'>... Aw, someone beat me to this one.</span>")
+				P.gib()
```
Comment (bobbahbrown) [inline code/modules/mob/living/simple_animal/friendly/dog.dm:538]: I think you've accidentally left these lines indented, with the new return these lines shouldn't be changed at all

Comment (itseasytosee) [reply code/modules/mob/living/simple_animal/friendly/dog.dm]: Yeah web edit gang

**Pull request closed without merging**